www/class/config-generate-remote/Relations/BrokerInfo.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/*
+ * Copyright 2005 - 2020 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+namespace ConfigGenerateRemote\Relations;
+
+use \PDO;
+use ConfigGenerateRemote\Abstracts\AbstractObject;
+
+class BrokerInfo extends AbstractObject
+{
+    private $useCache = 1;
+    private $doneCache = 0;
+
+    private $brokerInfoCache = [];
+
+    protected $table = 'cfg_centreonbroker_info';
+    protected $generateFilename = 'cfg_centreonbroker_info.infile';
+    protected $stmtBrokerInfo = null;
+
+    protected $attributesWrite = [
+        'config_id',
+        'config_key',
+        'config_value',
+        'config_group',
+        'config_group_id',
+        'grp_level',
+        'subgrp_id',
+        'parent_grp_id',
+        'fieldIndex'
+    ];
+
+    /**
+     * Constructor
+     *
+     * @param \Pimple\Container $dependencyInjector
+     */
+    public function __construct(\Pimple\Container $dependencyInjector)
+    {
+        parent::__construct($dependencyInjector);
+        $this->buildCache();
+    }
+
+    /**
+     * Build cache of broker info
+     *
+     * @return void
+     */
+    private function cacheBrokerInfo()
+    {
+        $stmt = $this->backendInstance->db->prepare(
+            "SELECT *
+             FROM cfg_centreonbroker_info"
+        );
+
+        $stmt->execute();
+        $values = $stmt->fetchAll(PDO::FETCH_ASSOC);
+        foreach ($values as &$value) {
+            if (!isset($this->brokerInfoCache[$value['config_id']])) {
+                $this->brokerInfoCache[$value['config_id']] = [];
+            }
+            $this->brokerInfoCache[$value['config_id']][] = $value;
+        }
+    }
+
+    /**
+     * Build cache
+     *
+     * @return void
+     */
+    private function buildCache()
+    {
+        if ($this->doneCache == 1) {
```

Comment:
why not use true/false ?

Reply:
same usage in other class. be consistent